cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/deploy/pipeline/AppDeploymentInfo.java

```diff
@@ -56,6 +56,8 @@ public class AppDeploymentInfo {
   private final ChangeDetail changeDetail;
   @Nullable
   private final ApplicationSpecification deployedApplicationSpec;
+  @Nullable
+  private final Boolean isUpgrade;
```

Comment:
@albertshau Is this approach fine? Setting this flag only during upgrades - that way we can distinguish between upgrades and other deploy events

Reply:
yes, though this doesn't need to be nullable, it can be a boolean.